src/GreenPipes/IContextSource.cs

```diff
@@ -0,0 +1,10 @@
+﻿using System.Threading;
+using System.Threading.Tasks;
+
+namespace GreenPipes
+{
+    public interface IContextSource<TContext> where TContext : class, PipeContext
+    {
+        Task Fill(IPipe<TContext> pipe, CancellationToken token = default(CancellationToken));

```

Comment:
I'm thinking we connect pipes to a source and then start up the source. The source seems like it should be configured independent of the pipes. sort of like how the bus is configured, each receive endpoint is configured, and the pipes are composed and then connected to it separately.